app/controllers/declaration/AdditionalActorsRemoveController.scala

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.declaration
+
+import controllers.actions.{AuthAction, JourneyAction}
+import controllers.navigation.Navigator
+import forms.common.YesNoAnswer
+import forms.common.YesNoAnswer.{form, YesNoAnswers}
+import forms.declaration.DeclarationAdditionalActors
+import javax.inject.Inject
+import models.declaration.DeclarationAdditionalActorsData
+import models.requests.JourneyRequest
+import models.{ExportsDeclaration, Mode}
+import play.api.data.Form
+import play.api.i18n.I18nSupport
+import play.api.mvc._
+import services.cache.ExportsCacheService
+import uk.gov.hmrc.play.bootstrap.controller.FrontendController
+import views.html.declaration.additionalActors.additional_actors_remove
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class AdditionalActorsRemoveController @Inject()(
+  authenticate: AuthAction,
+  journeyType: JourneyAction,
+  override val exportsCacheService: ExportsCacheService,
+  navigator: Navigator,
+  mcc: MessagesControllerComponents,
+  removePage: additional_actors_remove
+)(implicit ec: ExecutionContext)
+    extends FrontendController(mcc) with I18nSupport with ModelCacheable with SubmissionErrors {
+
+  def displayPage(mode: Mode, id: String): Action[AnyContent] = (authenticate andThen journeyType) { implicit request =>
```

Comment:
You didn't fancy the "generate id from index.hashcode" pattern then?
Your implementation will work, of course.  But it allows the user to alter the url which will then display and attempt to "remove" an actor that doesn't exist.

Reply:
We can alter the URL both ways. I guess this approach is more predictable and memorable.

Reply:
What I meant is that with your approach you can do things like this - 

http://localhost:6791/customs-declare-exports/declaration/authorisations-required/AEOC-STEVE_IS_GREAT/change

To produce this 

![image](https://user-images.githubusercontent.com/46934286/84988450-ea471280-b139-11ea-9609-4745512da899.png)

Does that matter?  I don't know.